### PR TITLE
RELATED: RAIL-4644 tooling to support run cypress test with both backend Bear and Tiger

### PR DESCRIPTION
**Pull request by tuan-tran-thanh**
---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by tuan-tran-thanh** [comment]
extended check plugins

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

File: libs/sdk-ui-tests-e2e/scripts/run_isolated_compose.js [hunk 1]
```diff
@@ -1,56 +1,103 @@
...
+                authorization = {
+                    token: TIGER_API_TOKEN,
+                };
+            }
         }
 
         if (recording) {
-            deleteRecordings(specFilesFilter);
-        } else if (!recordingsPresent()) {
+            // deleteRecordings(specFilesFilter, SDK_BACKEND, CYPRESS_TEST_TAGS.split(","));
```
Comment (no23reason) [inline libs/sdk-ui-tests-e2e/scripts/run_isolated_compose.js:98]: Should this be uncommented?

File: libs/sdk-ui-tests-e2e/scripts/run-scenarios-local.js [hunk 2]
```diff
@@ -10,16 +10,14 @@ async function main() {
...
-        "-e BACKEND_URL=https://staging3.intgdc.com",
+        "--add-host=host.docker.internal:host-gateway",
         "-v",
         `${process.env.PWD}/scenarios/build:/usr/share/nginx/html/gooddata-ui-sdk:ro`,
         "-v",
         `${process.env.PWD}/nginx/nginx.conf:/etc/nginx/nginx.conf:ro`,
         "-v",
         `${process.env.PWD}/nginx/proxy-isolated-tests-local.conf:/etc/nginx/extra-conf.d/proxy-isolated-tests-local.conf:ro`,
-        "020413372491.dkr.ecr.us-east-1.amazonaws.com/3rdparty/nginxinc/nginx-unprivileged:1.23.1-alpine",
+        "nginxinc/nginx-unprivileged:1.23.1-alpine",
```
Comment (no23reason) [inline libs/sdk-ui-tests-e2e/scripts/run-scenarios-local.js:20]: We should keep the ECR to avoid the rate limit problems potentially?

File: libs/sdk-ui-tests-e2e/package.json [hunk 3]
```diff
@@ -32,38 +32,46 @@
...
-        "validate": "npm run dep-cruiser && npm run eslint && npm run prettier-check",
-        "validate-ci": "npm run dep-cruiser-ci && npm run eslint-ci && npm run prettier-check",
-        "cypress": "node ./scripts/run_cypress_local.js",
-        "docker-build": "npm run build-scenarios && DOCKER_BUILDKIT=0 docker build --file Dockerfile -t gooddata-ui-sdk .",
+        "prettier-check": "prettier --check './**/*.{ts,js,cjs,tsx,jsx,json,scss,md,yaml,html}'",
+        "prettier-write": "prettier --write './**/*.{ts,js,cjs,tsx,jsx,json,scss,md,yaml,html}'",
+        "dep-cruiser": "depcruise --validate .dependency-cruiser.cjs --output-type err-long cypress/",
+        "dep-cruiser-ci": "depcruise --validate .dependency-cruiser.cjs --output-type err-long cypress/",
+        "validate": "yarn dep-cruiser && yarn eslint && yarn prettier-check",
+        "validate-ci": "yarn dep-cruiser-ci && yarn eslint-ci && yarn prettier-check",
```
Comment (no23reason) [inline libs/sdk-ui-tests-e2e/package.json:49]: Why did you change the `npm run XXX` to `yarn XXX`? We do not use `yarn` in the SDK repo at all. If it is not absolutely necesary please, use `npm run`

File: libs/sdk-ui-tests-e2e/Dockerfile [hunk 4]
```diff
@@ -1,19 +1,21 @@
...
 COPY . .
 
 ENV FORCE_COLOR 0
 
-FROM 020413372491.dkr.ecr.us-east-1.amazonaws.com/3rdparty/nginxinc/nginx-unprivileged:1.23.1-alpine
+RUN yarn install && yarn build-scenarios
 
-COPY --from=build-stage ./libs/sdk-ui-tests-e2e/nginx/nginx.conf /etc/nginx/nginx.conf
-COPY --from=build-stage ./libs/sdk-ui-tests-e2e/scenarios/build /usr/share/nginx/html/gooddata-ui-sdk/
+FROM nginxinc/nginx-unprivileged:1.23.1-alpine
```
Comment (no23reason) [inline libs/sdk-ui-tests-e2e/Dockerfile:11]: Should we keep the ECR?

File: libs/sdk-ui-tests-e2e/Dockerfile_local [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # (C) 2021 GoodData Corporation
 
-FROM 020413372491.dkr.ecr.us-east-1.amazonaws.com/3rdparty/nginxinc/nginx-unprivileged:1.23.1-alpine
+FROM nginxinc/nginx-unprivileged:1.23.1-alpine
```
Comment (no23reason) [inline libs/sdk-ui-tests-e2e/Dockerfile_local:3]: Should we keep the ECR?

File: libs/sdk-ui-tests-e2e/reference_workspace/lib/cleanup_tiger_workspaces.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        TIGER_API_TOKEN,
+        HOST,
+        SDK_BACKEND,
+    );
+}
+
+async function main() {
+    try {
+        // eslint-disable-next-line no-console
+        console.log("Clean up Expired Tiger Workspaces");
```
Comment (no23reason) [inline libs/sdk-ui-tests-e2e/reference_workspace/lib/cleanup_tiger_workspaces.js:30]: Should probably use only `log` like the other places in this file?

File: libs/sdk-ui-tests-e2e/scenarios/src/contexts/Auth/backend.ts [hunk 7]
```diff
@@ -1,23 +1,37 @@
...
+        return backend.withAuthentication(
+            new FixedLoginAndPasswordAuthProvider(process.env.USER_NAME!, process.env.PASSWORD!),
+        );
+    } else {
+        const sdkBackend: never = BACKEND_TYPE;
+        throw new Error(`unknown SDK backend: ${sdkBackend}`);
+    }
+}
 
 export const backendWithCredentials = (
```
Comment (no23reason) [inline libs/sdk-ui-tests-e2e/scenarios/src/contexts/Auth/backend.ts:31]: This function seems useless now, should we remove it?

File: libs/sdk-ui-tests-e2e/reference_workspace/lib/cleanup_tiger_workspaces.js (hunk 6 above)
Comment (tuan-tran-thanh) [reply libs/sdk-ui-tests-e2e/reference_workspace/lib/cleanup_tiger_workspaces.js:30]: fixed

Comment (tuan-tran-thanh) [reply libs/sdk-ui-tests-e2e/reference_workspace/lib/cleanup_tiger_workspaces.js:30]: fixed

File: libs/sdk-ui-tests-e2e/scripts/run_isolated_compose.js (hunk 1 above)
Comment (tuan-tran-thanh) [reply libs/sdk-ui-tests-e2e/scripts/run_isolated_compose.js:98]: fix

File: libs/sdk-ui-tests-e2e/Dockerfile (hunk 4 above)
Comment (tuan-tran-thanh) [reply libs/sdk-ui-tests-e2e/Dockerfile:11]: fix

File: libs/sdk-ui-tests-e2e/Dockerfile_local (hunk 5 above)
Comment (tuan-tran-thanh) [reply libs/sdk-ui-tests-e2e/Dockerfile_local:3]: fix

File: libs/sdk-ui-tests-e2e/scripts/run-scenarios-local.js (hunk 2 above)
Comment (tuan-tran-thanh) [reply libs/sdk-ui-tests-e2e/scripts/run-scenarios-local.js:20]: fixed

File: libs/sdk-ui-tests-e2e/package.json (hunk 3 above)
Comment (tuan-tran-thanh) [reply libs/sdk-ui-tests-e2e/package.json:49]: fixed

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10851/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12931/ : FAILURE in 59s

**Comment by caohhung** [comment]
extended check cypress

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10857/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/332/ : FAILURE in 5m 09s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10856/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12941/ : FAILURE in 10m 45s

**Comment by caohhung** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10859/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12944/ : FAILURE in 10m 46s

**Comment by caohhung** [comment]
extended check cypress

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by caohhung** [comment]
extended check cypress

**Comment by caohhung** [comment]
extended check cypress

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10865/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/333/ : FAILURE in 7m 50s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10864/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12951/ : FAILURE in 13m 21s

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10870/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/334/ : FAILURE in 5m 15s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by caohhung** [comment]
extended check cypress

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10874/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/335/ : FAILURE in 9m 12s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10873/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12960/ : FAILURE in 13m 10s

**Comment by caohhung** [comment]
extended check cypress

**Comment by gdgate** [comment]
Build in pipeline check-extended-cypress aborted.

**Comment by caohhung** [comment]
extended check cypress

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10879/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/337/ : FAILURE in 6m 28s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10878/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12964/ : FAILURE in 12m 07s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10897/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12982/ : FAILURE in 33m 19s

**Comment by ngocmanhha** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10898/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12983/ : FAILURE in 31m 39s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10899/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12985/ : SUCCESS in 21m 29s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10902/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12989/ : SUCCESS in 20m 04s

**Comment by tuan-tran-thanh** [comment]
extended check cypress

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 🎉 Great news! Tests were successful.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**dashboard.spec.ts**|9|-|55.5440s|
|✅|**attributeFilterButton.spec.ts**|9|-|29.4000s|
|✅|**pivotTableSizingReset.spec.ts**|15|-|38.1710s|
|✅|**pivotTableSizing.spec.ts**|1|-|4.6670s|
|✅|**attributeFilterButtonWithSelection.spec.ts**|2|-|6.6700s|
|✅|**dateFilter.spec.ts**|9|-|13.6000s|
|✅|**pivotTableRemoveOneAggregation.spec.ts**|1|-|6.6110s|
|✅|**barChartDrilling.spec.ts**|4|-|6.3120s|
|✅|**headlineChartDrilling.spec.ts**|2|-|3.5480s|
|✅|**pivotTableRemoveOneFromMutipleAggregation.spec.ts**|1|-|6.9380s|
|✅|**pivotTableRemoveAllFromMutipleAggregation.spec.ts**|1|-|6.5370s|
|✅|**parentChildFiltersWithPlaceholders.spec.ts**|3|-|16.8400s|
|✅|**pivotTableAggregationsMenuAndAddTotal.spec.ts**|5|-|19.3030s|
|✅|**attributeFilterButtonParentChild.spec.ts**|3|-|13.9370s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-docker/344/).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10903/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-docker/344/ : SUCCESS in 11m 55s

File: common/scripts/ci/run_cypress_isolated_tests.sh [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+_RUSHX="${DIR}/docker_rushx.sh"
+
+# : "${SDK_BACKEND:?}"  # Todo: To be enabled later
+
+sdk_backend=$(tr <<< $SDK_BACKEND '[:upper:]' '[:lower:]')
+
+pushd $E2E_TEST_DIR
+cat > .env <<-EOF
+SDK_BACKEND=${SDK_BACKEND:-BEAR}
+TEST_WORKSPACE_ID=c76e0537d0614abb0027f7c992656b964922506f
```
Comment (tuan-tran-thanh) [inline common/scripts/ci/run_cypress_isolated_tests.sh:18]: Can we use[ this way](https://github.com/gooddata/gooddata-ui-sdk/pull/3321/files#diff-2c4178351f6248c4e681d1fdfbcb1f77cc0b221e44f64138349718e89f199b2fR62) to add value of TEST_WORKSPACE_ID to `.env`

File: common/scripts/ci/run_cypress_recording.sh [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+        -v $ROOT_DIR:/workspace \
+        $CYPRESS_IMAGE $@
+}
+
+echo "⭐️ 2/8 Run rush install / build"
+$_RUSH install
+$_RUSH build -t sdk-ui-tests-e2e
+
+echo "⭐️ 3/8 create reference workspace on the SDK_BACKEND=$SDK_BACKEND"
+ _docker_run node reference_workspace/create_ref_workspace.js
```
Comment (tuan-tran-thanh) [inline common/scripts/ci/run_cypress_recording.sh:67]: Can we use `_RUSHX libs/sdk-ui-tests-e2e create-ref-workspace`?

File: common/scripts/ci/run_cypress_recording.sh [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+mkdir -p ./recordings/mappings/$SDK_BACKEND
+docker build --file Dockerfile_local -t $IMAGE_ID . || exit 1
+
+echo "⭐️ 6/8 Run isolated recording against TEST_BACKEND=$TEST_BACKEND."
+MODE=record NO_COLOR=1 docker-compose -f docker-compose-isolated.yaml up \
+    --abort-on-container-exit --exit-code-from isolated-tests \
+    --force-recreate --always-recreate-deps --renew-anon-volumes --no-color
+
+echo "⭐️ 7/8 delete reference workspace on the host"
+ _docker_run node reference_workspace/delete_ref_workspace.js
```
Comment (tuan-tran-thanh) [inline common/scripts/ci/run_cypress_recording.sh:86]: Can we use `_RUSHX libs/sdk-ui-tests-e2e delete-ref-workspace`?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10919/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13011/ : SUCCESS in 21m 02s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10920/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13012/ : SUCCESS in 20m 42s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3321

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/217/ : FAILURE in 2m 53s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/3483/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/2332/ : SUCCESS in 17m 29s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13016/ : SUCCESS in 26m 16s
